lib/chai/utils/addChainableMethod.js

```diff
@@ -75,13 +75,18 @@ module.exports = function (ctx, name, method, chainingBehavior) {
   ctx.__methods[name] = chainableBehavior;
 
   Object.defineProperty(ctx, name,
-    { get: function () {
+    { get: function chainableMethodGetter() {
         chainableBehavior.chainingBehavior.call(this);
 
-        var assert = function assert() {
-          var old_ssfi = flag(this, 'ssfi');
-          if (old_ssfi)
-            flag(this, 'ssfi', assert);
+        var chainableMethodWrapper = function () {
+          // Use this chainable method wrapper as the starting point for
+          // removing implementation frames from the stack trace of a failed
+          // assertion. Note that this is the correct starting point even if
+          // this assertion has been overwritten since overwriting a chainable
+          // method merely replaces the saved methods in `ctx.__methods` instead
+          // of completely replacing the overwritten assertion.
+          flag(this, 'ssfi', chainableMethodWrapper);
```

Comment:
Hmm, since it's pretty specific I think there's nothing we can do about it.
Let's leave it that way.

Maybe a comment indicating what it means somewhere would also be nice, especially if we added a link to that comment, which was really useful for me when understanding how the whole stack trace manipulation works.